zqd/storage/archivestore/archivestore.go

```diff
@@ -40,8 +41,9 @@ func (s *Storage) Open(ctx context.Context, span nano.Span) (zbuf.ReadCloser, er
 		}
 	}()
 	zctx := resolver.NewContext()
-	err = archive.SpanWalk(s.ark, func(sp nano.Span, zngpath string) error {
-		if !span.Overlaps(sp) {
+	err = archive.SpanWalk(s.ark, func(si archive.SpanInfo, zardir string) error {
+		zngpath := archive.ZarDirToLog(zardir)
```

Comment:
Nit: Move zngpath's declaration adjacent to its use. Or better yet, drop it and just say `f, err := fs.Open(archive.ZarDirToLog(zardir)`.